### PR TITLE
Tweak some ‘expected…’ error messages

### LINKED ISSUE #21153: "Expected `fn`, found `something`" should say "Expected `fn` or `type`, found `something`"
If you are trying to write a new trait that should look like this:

```rust
trait MyTrait<T> : Iterator {
    type Item = T
    ...
}
```

And you leave out the word `type`, like in [this example](http://is.gd/gKGi7A), you get the error message
```
error: expected `fn`, found `Item`
```

Even though the missing word is `type`, not `fn`.

### PR DESCRIPTION
**Pull request by ftxqxd**
Fixes #21153.

### CODE REVIEW

File: src/libsyntax/parse/parser.rs [hunk 1]
```diff
@@ -574,9 +579,27 @@ impl<'a> Parser<'a> {
         is_present
     }
 
+    pub fn check_keyword(&mut self, kw: keywords::Keyword) -> bool {
+        self.expected_tokens.push(TokenType::Keyword(kw));
+        if self.token.is_keyword(kw) {
+            true
+        } else {
+            false
+        }
```
Comment (goffrie) [inline src/libsyntax/parse/parser.rs]: Why not just `self.token.is_keyword(kw)`?

Comment (P1start) [inline src/libsyntax/parse/parser.rs]: Whoops! Thanks for pointing this out. Fix incoming…

**Comment by huonw** [comment]
Thanks for this; r=me with a rebase (sorry for the delay :( ).

**Comment by P1start** [comment]
@huonw Rebased.

**Comment by alexcrichton** [comment]
@bors: retry

Sorry for the noise :(

**Pull request merged**